src/lib.rs

```diff
@@ -1195,6 +1195,66 @@ fn graph_adjacency_matrix(
     Ok(matrix.into_pyarray(py).into())
 }
 
+#[pyfunction]
+fn graph_all_simple_paths(
+    graph: &graph::PyGraph,
+    from: usize,
+    to: usize,
+    min_depth: Option<usize>,
+    cutoff: Option<usize>,
+) -> PyResult<Vec<Vec<usize>>> {
+    let from_index = NodeIndex::new(from);
+    let to_index = NodeIndex::new(to);
+    let min_intermediate_nodes: usize = match min_depth {
+        Some(depth) => depth - 2,
+        None => 0,
+    };
+    let cutoff_petgraph: Option<usize> = match cutoff {
+        Some(depth) => Some(depth - 2),
+        None => None,
+    };
+    let result: Vec<Vec<usize>> = algo::all_simple_paths(
+        graph,
+        from_index,
+        to_index,
+        min_intermediate_nodes,
+        cutoff_petgraph,
+    )
+    .map(|v: Vec<NodeIndex>| v.into_iter().map(|i| i.index()).collect())
+    .collect();
+    Ok(result)
+}
+
+#[pyfunction]
+fn dag_all_simple_paths(
+    graph: &PyDAG,
+    from: usize,
+    to: usize,
+    min_depth: Option<usize>,
+    cutoff: Option<usize>,
+) -> PyResult<Vec<Vec<usize>>> {
+    let from_index = NodeIndex::new(from);
+    let to_index = NodeIndex::new(to);
+    let min_intermediate_nodes: usize = match min_depth {
+        Some(depth) => depth - 2,
+        None => 0,
+    };
+    let cutoff_petgraph: Option<usize> = match cutoff {
+        Some(depth) => Some(depth - 2),
+        None => None,
+    };
+    let result: Vec<Vec<usize>> = algo::all_simple_paths(
```

Comment:
this is provided by petgraph?

Reply:
Yeah, this is a built in algorithm in petgraph: https://docs.rs/petgraph/0.5.1/petgraph/algo/fn.all_simple_paths.html